### DISCUSSION TITLE
Curious about GitRepositories per app

### DISCUSSION THREAD

**Discussion by mnarrell** [onedr0p/cluster-template#720] [Q&A] [answered]
I'm interested in your thoughts around having a GitRepository for each of the applications you're deploying.  I realize that certain standalone manifests might not support dependencies, but are there other advantages than having a single GitRepository and apps enabled with Kustomizations and HelmReleases?

**Comment by onedr0p** [comment]
Having a GitRepo resource for each of the things in the same repo on the same branch seems to me to be very inefficient. Personally I don't think that would be a good idea, but what make you think about this? Are you trying to solve a problem?

**Reply by mnarrell** [reply]
Maybe I'm misunderstanding.  My question is about resources like this: https://github.com/onedr0p/flux-cluster-template/blob/main/kubernetes/apps/cert-manager/cert-manager/ks.yaml#L11

Is there a reason or preference to using this, over just a kustomization file pointing to a HelmRelease and the issuers manifests?

**Reply by onedr0p** [reply, marked as answer]
Ah you said GitRepository not Flux Kustomizations. This is done to have proper resource ordering since you cannot deploy a cert-manager `ClusterIssuer` without the CRDs installed first.

**Reply by mnarrell** [reply]
Thanks
